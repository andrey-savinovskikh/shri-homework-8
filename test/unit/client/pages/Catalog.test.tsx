import React from 'react';
import renderer from 'react-test-renderer';
import "@testing-library/jest-dom/extend-expect";
import {screen, waitFor, render} from '@testing-library/react';
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";

import {PRODUCTS} from "../../../utils/mockApi";
import {CartState} from "../../../../src/common/types";
import {getStore} from "../../../utils/getStore";
import {Catalog} from "../../../../src/client/pages/Catalog";

const getCatalog = (initialCartData: CartState = {}) => {
    const store = getStore(initialCartData);

    const catalog = (
        <BrowserRouter>
            <Provider store={store}>
                <Catalog/>
            </Provider>
        </BrowserRouter>
    )

    return {store, catalog}
}

describe('Страница Catalog', () => {
    it('Корректный снапшот страницы без товаров', () => {
        const {catalog} = getCatalog();
        const tree = renderer.create(catalog).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('У товаров отображаются коректные данные', async () => {
        const {catalog} = getCatalog();
        render(catalog);

        await waitFor(() => {
            PRODUCTS.every((product, index) => {
                const item = screen.getAllByTestId(product.id)[1];

                const price = item.querySelector('.ProductItem-Price').innerHTML,
                    name = item.querySelector('.ProductItem-Name').innerHTML,
                    link = item.querySelector('.ProductItem-DetailsLink').getAttribute('href');

                expect(price.includes(String(product.price))).toBeTruthy();
                expect(name === product.name).toBeTruthy();
                expect(link === `/catalog/${product.id}`).toBeTruthy();
            });
        });
    });
})