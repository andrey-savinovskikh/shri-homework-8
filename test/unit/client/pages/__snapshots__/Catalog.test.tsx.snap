// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Страница Catalog Корректный снапшот страницы без товаров 1`] = `
<div
  className="Catalog"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h1>
        Catalog
      </h1>
    </div>
  </div>
  <div
    className="row"
  >
    LOADING
  </div>
</div>
`;
